README.md

```diff
@@ -214,6 +215,22 @@ jobs:
       - uses: actions/checkout@v2
 ```
 
+## Checkout pull request on `pull_request_target`
+
+```yaml
+on:
+  - pull_request_target
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: refs/pull/${{ github.event.pull_request.number }}/head
+```
+
+**WARNING! NEVER** run code from pull requests of public repositories! The token of `pull_request_target` event has write access.
```

Comment:
I do not agree with this general warning. `pull_request_target` uses the workflow of the target, so no one can modify it from a pull request and include a secret this way. There can only be a security leak if the workflow leaks it to the checked out source.

Except `GITHUB_TOKEN`, secrets are not available to runners unless they are explicitly passed. The token itself is limited to [read access](https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#permissions-for-the-github_token) on forked repositories. The token expires after each workflow run and doesn't grant access beyond the repository itself. So there is no issue with the token.

It should be clear to the maintainer of a repository that any other secret that it passed to executables of the checked-out code can be used maliciously. This only affects executed code. One might perform some check or optimization using an action that receives the code and it would still be secure. Of course this depends on the level of trust into the third party that created the action. But this issue exists in general, regardless of event type.

Reply:
@matfax The `pull_request_target` runs in the context of the base repository of the pull request, i.e., its token has read/write access even for pull requests from forked repos.

Reply:
I've tested various things now. Even though GITHUB_TOKEN doesn't seem to be available all the time, the `github.token` from the context has indeed write permissions as you stated. actions/checkout sets this context token as repository token, so it's available anyways once checkout is used. I find it rather odd that GitHub introduced `pull_request_target` as some kind of safe alternative but how are you supposed to check anything in a pull request if you can't even check out its code? Imho, they should change the permissions of the token if used in `pull_request_target` to read only the repository + write access to the pull request (except merge). The use case of checking forked pull requests is integrated and documented insufficiently.